internal/pkg/agent/install/uninstall.go

```diff
@@ -99,27 +99,23 @@ func Uninstall(cfgFile, topPath, uninstallToken string) error {
 func RemovePath(path string) error {
 	const arbitraryTimeout = 5 * time.Second
 	start := time.Now()
-	nextSleep := 1 * time.Millisecond
-	for {
-		err := os.RemoveAll(path)
-		if err == nil {
-			return nil
-		}
-		if isBlockingOnExe(err) {
-			// try to remove the blocking exe
-			err = removeBlockingExe(err)
-		}
-		if err == nil {
-			return nil
-		}
-		if !isRetryableError(err) {
-			return err
+	var lastErr error
+	for time.Since(start) <= arbitraryTimeout {
+		lastErr = os.RemoveAll(path)
+
+		if lastErr == nil || !isRetryableError(lastErr) {
+			return lastErr
 		}
 
-		if d := time.Since(start) + nextSleep; d >= arbitraryTimeout {
-			return err
+		if isBlockingOnExe(lastErr) {
+			// try to remove the blocking exe and try again to clean up the path
+			_ = removeBlockingExe(lastErr)
 		}
+
+		time.Sleep(50 * time.Millisecond)
```

Comment:
I think it might be too much pressure on the OS. But it's a guess. However I think a counter of how many tries happened would be good.

Reply:
50 milliseconds is a long time to a computer, this is an improvement over what it did before which was not sleeping.

I'd be in favour of logging how long this function took to complete so we can tell if any future flakiness is just related to marginal timing.

Reply:
https://github.com/elastic/elastic-agent/issues/3434